pkg/connector/kairosdb.go

```diff
@@ -0,0 +1,494 @@
+		   			return nil, fmt.Errorf("unable to compile regexp pattern: %s", err)
+		   		}
+		   		if connector.metricRe, err = compileRePattern(mp); err != nil {
+		   			return nil, fmt.Errorf("unable to compile regexp pattern: %s", err)
+		   		}
+		*/
+
+		// Enforce minimal timeout value bound
+		if connector.timeout <= 0 {
+			connector.timeout = kairosdbDefaultTimeout
+		}
+
+		return connector, nil
+	}
+}
+
+// GetName returns the name of the current connector.
+func (connector *KairosdbConnector) GetName() string {
+	return connector.name
+}
+
+// GetPlots retrieves time series data from provider based on a query and a time interval.
+func (connector *KairosdbConnector) GetPlots(query *plot.Query) ([]plot.Series, error) {
+	var resultSeries []plot.Series
+
+	if len(query.Series) == 0 {
+		return nil, fmt.Errorf("kairosdb[%s]: requested series list is empty", connector.name)
+	}
+
+	JSONquery, err := kairosdbBuildJSONQuery(query, connector.series)
+	if err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to build or marshal JSON query: %s", connector.name, err)
+	}
+
+	httpTransport := &http.Transport{
+		Dial: (&net.Dialer{
+			// Enable dual IPv4/IPv6 stack connectivity:
+			DualStack: true,
+			// Enforce HTTP connection timeout:
+			Timeout: time.Duration(connector.timeout) * time.Second,
+		}).Dial,
+	}
+	if connector.insecureTLS {
+		httpTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	httpClient := http.Client{Transport: httpTransport}
+
+	request, err := http.NewRequest("POST", strings.TrimSuffix(connector.URL, "/")+kairosdbURLQueryMetric, bytes.NewBuffer(JSONquery))
+	if err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to set up HTTP request: %s", connector.name, err)
+	}
+
+	request.Header.Add("User-Agent", "Facette")
+	request.Header.Add("X-Requested-With", "KairosdbConnector")
+	request.Header.Set("Content-Type", "application/json")
+
+	response, err := httpClient.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to perform HTTP request: %s", connector.name, err)
+	}
+	defer response.Body.Close()
+
+	if err = kairosdbCheckBackendResponse(response); err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: invalid HTTP backend response: %s", connector.name, err)
+	}
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to read HTTP response body: %s", connector.name, err)
+	}
+
+	var JSONresponse map[string][]MetricsQueryResponse
+	if err = json.Unmarshal(data, &JSONresponse); err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to unmarshal JSON data: %s", connector.name, err)
+	}
+
+	if resultSeries, err = kairosdbExtractPlots(query, connector.series, JSONresponse["queries"]); err != nil {
+		return nil, fmt.Errorf("kairosdb[%s]: unable to extract plot values from backend response: %s", connector.name, err)
+	}
+
+	// return nil, fmt.Errorf("kairosdb[%s]: not implemented, yet.", connector.name)
```

Comment:
Am 18.01.2015 um 12:37 schrieb Marc Falzon:
> In pkg/connector/kairosdb.go
> <https://github.com/facette/facette/pull/153#discussion-diff-23132676>:
> 
>> +	// return nil, fmt.Errorf("kairosdb[%s]: not implemented, yet.", connector.name)
> 
> Artefact from early development stage?

Yepp. Should be removed in next commit.